packages/manager/modules/pci/src/projects/new/components/voucher/controller.js

```diff
@@ -55,13 +59,32 @@ export default class PciProjectNewVoucherCtrl {
 
   onVoucherFormSubmit() {
     this.loading.check = true;
+    this.globalLoading.isVoucherValidating = true;
+    this.disablePaymentMethods(true);
 
     return this.checkVoucherValidity(this.model.voucher.value)
       .then((eligibilityOpts) => {
+        this.model.isVoucherRequirePaymentMethod =
+          eligibilityOpts.voucher.paymentMethodRequired;
         this.model.voucher.setInfos(eligibilityOpts.voucher);
+        this.eligibility.setOptions(eligibilityOpts);
         this.setVoucherFormState();
 
-        this.eligibility.setOptions(eligibilityOpts);
+        return eligibilityOpts;
+      })
+      .then((eligibilityOpts) => {
+        if (eligibilityOpts.voucher?.error) {
+          return Promise.reject(eligibilityOpts);
```

Comment:
For the moment, as the application is still under AngularJS:
```suggestion
          return this.$q.reject(eligibilityOpts);
```